### ISSUE TITLE
Scaling logic in compute_maxrl_outcome_advantage vs paper derivation

### ISSUE THREAD

**Issue by marviss** [tajwarfahim/maxrl#1]
Greetings and thank you for providing this code base! 

While looking into the implementation of the `MaxRL` estimator, I was hoping to clarify the scaling logic used in `core_algos.py`. Specifically, the current implementation appears to omit the $1/N$ normalization factor (where $N$ is the number of responses per prompt) defined in the original algorithm (Tajwar et al., 2024). I’m interested in whether this deviation from the paper's scaling was a deliberate design choice for this framework.

**Reference Equation (Tajwar et al., Algorithm 1, Step 8):**
$$\hat{g}(x) = \frac{1}{N \hat{r}(x)} \sum_{j=1}^N (r_j - \hat{r}(x)) \nabla \log \pi(y_j|x)$$

The `compute_maxrl_outcome_advantage` [function](https://github.com/tajwarfahim/maxrl/blob/6340306bef68fc8ee2ffe9cc554b9ee256cebba1/verl/trainer/ppo/core_algos.py#L437) currently computes:

$$\text{Advantage} = \frac{r_j - \hat{r}(x)}{\hat{r}(x) + \epsilon}$$:

Relevant code part:
```
id2mean[idx] = torch.mean(torch.tensor(id2score[idx]))
for i in range(bsz):
    scores[i] = (scores[i] - id2mean[index[i]]) / (id2mean[index[i]] + epsilon)
```

**Question regarding the implementation:**
I would kindly like to clarify if the omission of the $1/N$ factor in the denominator was an intentional design choice for this implementation. More specifically:
- Is it expected that the user should manually scale down their learning rate by a factor of $N$ when using the `MaxRL` estimator compared to standard `GRPO`?
- Or, is this normalization intended to be handled by the global loss aggregation? 

Checking the [loss function](https://github.com/tajwarfahim/maxrl/blob/6340306bef68fc8ee2ffe9cc554b9ee256cebba1/verl/trainer/ppo/core_algos.py#L1199), it appears to me that the advantages are passed directly without further per-group scaling. Could the authors kindly help clarify this?

Thank you and kind regards!

**Comment by tajwarfahim** [comment]
Hi, 

We generally have N fixed across different prompts (the same number of rollouts is generated for each prompt), in which case the 1/N factor is a constant. Explicitly normalizing by 1/N has usually no effect, since it is a constant, and hence will be absorbed into the learning rate (which Adam will generally normalize anyway).

One can explicitly normalize by N, but other than some effect on the optimal hyper-parameters (learning rate), I don't think there would be any other difference.

In case you find substantial difference in behavior after explicitly normalizing by N or not, please let us know by opening a new ticket, and we would be happy to update the codebase + paper.

Thanks a lot!
